lib/Conversion/TorchToLinalg/TorchToLinalg.cpp

```diff
@@ -909,30 +1003,414 @@ class ConvertAtenNativeLayerNormOp
         rewriter
             .create<linalg::GenericOp>(
                 loc, initLayerNormTensor.getType(),
-                ValueRange{input, mean, var, weight, bias}, initLayerNormTensor,
+                ValueRange{input, mean, invSTD, weight, bias},
+                initLayerNormTensor,
                 /*indexingMaps=*/indexingMaps,
                 /*iteratorTypes=*/layerNormIterationTypes,
                 [&](OpBuilder &b, Location loc, ValueRange args) {
-                  Value input = args[0], mean = args[1], var = args[2],
+                  Value input = args[0], mean = args[1], invSTD = args[2],
                         weight = args[3], bias = args[4];
                   Value result = createLinalgPayloadCalculationForNormOps(
-                      b, loc, elemTy, input, mean, var, eps, weight, bias);
+                      b, loc, elemTy, input, mean, invSTD, eps, weight, bias);
                   b.create<linalg::YieldOp>(loc, result);
                 })
             .getResult(0);
+    SmallVector<bool> meanVarBroadcastMask(inputRank, false);
+    for (unsigned i = 0; i < inputRank; i++) {
+      if (i >= meanAndVarShapeRank)
+        meanVarBroadcastMask[i] = true;
+    }
+    Value meanResult = broadcast(loc, mean, meanAndVarShapeSizes,
+                                 meanVarBroadcastMask, rewriter, context);
+    Value varResult = broadcast(loc, invSTD, meanAndVarShapeSizes,
+                                meanVarBroadcastMask, rewriter, context);
     Type layerNormResultType = getTypeConverter()->convertType(op.getType(0));
     Type meanResultType = getTypeConverter()->convertType(op.getType(1));
     Type varResultType = getTypeConverter()->convertType(op.getType(2));
     Value layerNorm_ =
         rewriter.create<tensor::CastOp>(loc, layerNormResultType, layerNorm);
-    Value mean_ = rewriter.create<tensor::CastOp>(loc, meanResultType, mean);
-    Value var_ = rewriter.create<tensor::CastOp>(loc, varResultType, var);
+    Value mean_ =
+        rewriter.create<tensor::CastOp>(loc, meanResultType, meanResult);
+    Value var_ = rewriter.create<tensor::CastOp>(loc, varResultType, varResult);
     rewriter.replaceOp(op, {layerNorm_, mean_, var_});
     return success();
   }
 };
 } // namespace
 
+namespace {
```

Comment:
Please give brief pseudocode of the conversion. This will make reviewing the code easier.